runelite-client/src/main/java/net/runelite/client/plugins/StatusBars/StatusBarsOverlay.java

```diff
@@ -0,0 +1,314 @@
+import net.runelite.client.ui.overlay.OverlayLayer;
+import javax.inject.Inject;
+import net.runelite.client.game.SkillIconManager;
+import net.runelite.client.util.QueryRunner;
+
+@Slf4j
+public class StatusBarsOverlay extends Overlay
+{
+	// Static colors
+	private static final Color DEPLETED = new Color(25, 25, 25, 150);
+	private static final Color PRAYER_COLOR = new Color(50, 200, 200, 200);
+	private static final Color QUICK_PRAYER_COLOR = new Color(57, 255, 186, 225);
+	private static final Color BACKGROUND = new Color(0, 0 , 0, 150);
+	private static final Color HEALTH_COLOR = new Color(200, 35, 0, 150);
+	private static final Color POISONED_COLOR = new Color(0, 145, 0, 150);
+	private static final Color VENOMED_COLOR = new Color(0, 65, 0, 150);
+	private static final Color COUNTER_COLOR = new Color(255, 255,  255, 255);
+	private static final Color NO_COLOR = new Color(0, 0, 0, 0);
+	private static final Color HEAL_COLOR = new Color (255, 112, 6, 150);
+	private static final Color PRAYER_HEAL_COLOR = new Color(57, 255, 186, 75);
+	private static final int HEIGHT = 252;
+	private static final int WIDTH = 20;
+	private static final int PADDING = 1;
+	private static final int HEALTH_LOCATION_X = 2;
+	private static final int PRAYER_LOCATION_X = 3;
+	private final Client client;
+	private final StatusBarsConfig config;
+	private final SkillIconManager skillIconManager;
+	private final QueryRunner queryRunner;
+	private final ItemStatChanges itemStatChanges;
+
+	@Inject
+	private StatusBarsOverlay(Client client, StatusBarsConfig config, SkillIconManager skillIconManager, QueryRunner queryRunner, ItemStatChanges itemStatChanges)
+	{
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.client = client;
+		this.config = config;
+		this.skillIconManager = skillIconManager;
+		this.queryRunner = queryRunner;
+		this.itemStatChanges = itemStatChanges;
+	}
+
+	@Override
+	public Dimension render(Graphics2D g)
+	{
+		//Variables
+		int maxHealth = client.getRealSkillLevel(Skill.HITPOINTS);
+		int maxPrayer = client.getRealSkillLevel(Skill.PRAYER);
+		int currentHealth = client.getBoostedSkillLevel(Skill.HITPOINTS);
+		int currentPrayer = client.getBoostedSkillLevel(Skill.PRAYER);
+		int poisonState = client.getVar(VarPlayer.IS_POISONED);
+		int counterHealth = client.getBoostedSkillLevel(Skill.HITPOINTS);
+		int counterPrayer = client.getBoostedSkillLevel(Skill.PRAYER);
+		int quickPrayerState = client.getVar(Varbits.QUICK_PRAYER);
+		int imageSize = 16;
+		int prayerHealValue = 0 ;
+		int foodHealValue = 0 ;
+		Color healthBar;
+		Color prayerBar;
+		Color counter;
+		Widget widgetBankTitleBar = client.getWidget(WidgetInfo.BANK_TITLE_BAR);
+
+		//Store images
+		BufferedImage healthImage = skillIconManager.getSkillImage(Skill.HITPOINTS, true);
+		BufferedImage prayerImage = skillIconManager.getSkillImage(Skill.PRAYER, false);
+		BufferedImage noImage = new BufferedImage(17, 17, BufferedImage.TYPE_INT_ARGB);
```

Comment:
Dont create new buffered images every render cycle. Store the noImage as constant somewhere